src/web/mjs/connectors/mangaz.mjs

```diff
@@ -0,0 +1,90 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class Mangaz extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'mangaz';
+        super.label = 'Manga Library Z (マンガ図書館Z)';
+        this.tags = ['manga', 'japanese'];
+        this.url = 'https://www.mangaz.com';
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for(let page = 0, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const request = new Request(new URL('/title/addpage_renewal?query=&page='+page, this.url), {
+            method:'GET',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
+            }
+        });
+        const data = await this.fetchDOM(request, 'h4 > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'li.title');
+        const id = uri.pathname + uri.search;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getPages(chapter) {
+        const script = `
+            new Promise(async (resolve,reject) => {
+                let g = JCOMI.namespace("JCOMI.document")
+                let b = g.getDoc()
+                let img = g.getImages().map(ele => g.getLocationDir('enc') + ele.file + "?vw=" + encodeURIComponent(JCOMI.namespace("JCOMI.config").getVersion()))
+                resolve({img:img,b:b});
+            });
+        `;
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        const data = await Engine.Request.fetchUI(request, script)
+        return data.img.map(ele => this.createConnectorURI({
+            url:this.getAbsolutePath(ele, request.url),
+            key:data.b.Enc.key,
+            iv:data.b.Enc.iv
+        }));
+    }
+
+    async _handleConnectorURI(payload) {
+        let response = await fetch(payload.url)
+        let encrypted = await response.text();
```

Comment:
i changed from .arrayBuffer() to .text() because i get literally nothing with .arrayBuffer() after the decrypt 

![](https://i.imgur.com/QGrJXgd.png)

Reply:
note: encrypted.arrayBuffer()
![](https://i.imgur.com/yk1OFqC.png)
encrypted.text()
![](https://i.imgur.com/afJo1f3.png)

Reply:
You need to convert `ArrayBuffer` into `WordArray` or something else that CryptoJS understands.
I'm not used to CryptoJS, so i cannot help any further with that... all i can say that the parameters needs to be converted to a type that works with CryptoJS and the decrypted results needs to be converted from `WordArray` to `Uint8array`

WordArrays can be created from various sources, e.g.
CryptoJS.enc.Base64.parse(...)
CryptoJS.enc.Hex.parse(...)

Reply:
i have tried every combination of every different way used by anyone on google and the CryptoJS docs, i have no idea how to get anything without forge